src/commands/appSettings/AzureWebJobsStoragePromptStep.ts

```diff
@@ -6,52 +6,64 @@
-        const skipForNow: MessageItem = { title: localize('skipForNow', 'Skip for now') };
+        const connectStorageButton: MessageItem = { title: localize('connectStorageAccount', 'Connect Storage Account') };
+        const useEmulatorButton: MessageItem = { title: useEmulator };
+        const skipForNowButton: MessageItem = { title: skipForNow };
 
-        const message: string = localize('selectAzureWebJobsStorage', 'In order to debug, you must select a storage account for internal use by the Azure Functions runtime.');
+        const message: string = localize('connectAzureWebJobsStorage', 'In order to proceed, you must connect a storage account for internal use by the Azure Functions runtime.');
 
-        const buttons: MessageItem[] = [selectAccount];
-        if (process.platform === 'win32') {
-            // Only show on Windows until Azurite is officially supported: https://github.com/Azure/azure-functions-core-tools/issues/1247
-            buttons.push(useEmulator);
-        }
-        if (!this._suppressSkipForNow) {
-            buttons.push(skipForNow);
+        const buttons: MessageItem[] = [connectStorageButton, useEmulatorButton];
+
+        if (!this._options?.suppressSkipForNow) {
+            buttons.push(skipForNowButton);
         }
 
         const result: MessageItem = await context.ui.showWarningMessage(message, { modal: true }, ...buttons);
-        if (result === selectAccount) {
-            context.azureWebJobsStorageType = 'azure';
-        } else if (result === useEmulator) {
-            context.azureWebJobsStorageType = 'emulator';
+        if (result === connectStorageButton) {
+            context.azureWebJobsStorageType = ConnectionType.Azure;
+        } else if (result === useEmulatorButton) {
+            context.azureWebJobsStorageType = ConnectionType.NonAzure;
+        } else {
+            context.azureWebJobsStorageType = ConnectionType.None;
         }
 
-        context.telemetry.properties.azureWebJobsStorageType = context.azureWebJobsStorageType || 'skipForNow';
+        context.telemetry.properties.azureWebJobsStorageType = context.azureWebJobsStorageType;
     }
 
-    public shouldPrompt(context: T): boolean {
-        // Only should prompt if no storage account was selected
-        context.azureWebJobsStorageType = (!!context.storageAccount || !!context.newStorageAccountName) ? 'azure' : undefined;
+    public shouldPrompt(context: T & { eventHubConnectionType?: ConnectionTypeValues, sqlDbConnectionType?: ConnectionTypeValues }): boolean {
+        if (this._options?.preSelectedConnectionType) {
+            context.azureWebJobsStorageType = this._options.preSelectedConnectionType;
+        } else if (!!context.storageAccount || !!context.newStorageAccountName) {
+            context.azureWebJobsStorageType = ConnectionType.Azure;  // Only should prompt if no storage account was selected
```

Comment:
This comment should be moved to the top and updated, explaining what scenarios stifle the prompt.